react-instantsearch-hooks/geo-search/src/Airports.tsx

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useRef } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { Marker, Popup, useMapEvents } from 'react-leaflet';
+import { DivIcon } from 'leaflet';
+import { useConnector, useSearchBox } from 'react-instantsearch-hooks-web';
+import connectGeoSearch, {
+  GeoHit,
+  GeoSearchConnectorParams,
+  GeoSearchWidgetDescription,
+} from 'instantsearch.js/es/connectors/geo-search/connectGeoSearch';
+
+type UseGeoSearchProps = GeoSearchConnectorParams;
+
+function useGeoSearch(props?: UseGeoSearchProps) {
+  return useConnector<GeoSearchConnectorParams, GeoSearchWidgetDescription>(
+    connectGeoSearch,
+    props
+  );
+}
+
+export function Airports() {
+  const skipViewEffect = useRef(false);
+
+  const { query, refine: refineQuery } = useSearchBox();
+  const {
+    items,
+    refine: refineItems,
+    currentRefinement,
+    clearMapRefinement,
+  } = useGeoSearch();
+
+  const onViewChange = ({ target }) => {
+    skipViewEffect.current = true;
+
+    if (query.length > 0) {
+      refineQuery('');
+    }
+
+    refineItems({
+      northEast: target.getBounds().getNorthEast(),
+      southWest: target.getBounds().getSouthWest(),
+    });
+  };
+
+  const map = useMapEvents({
+    zoomend: onViewChange,
+    dragend: onViewChange,
+  });
+
+  useEffect(() => {
+    if (currentRefinement) {
+      clearMapRefinement();
+    }
+
+    if (items.length > 0 && !skipViewEffect.current) {
+      map.setView(items[0]._geoloc);
+    }
+
+    skipViewEffect.current = false;
+  }, [query]);
```

Comment:
React Hook useEffect has missing dependencies: 'clearMapRefinement', 'currentRefinement', 'items', and 'map'. Either include them or remove the dependency array. (react-hooks/exhaustive-deps)eslint

should this be fixed or ignored?

Reply:
I removed the need for `useEffect` since now the behavior is done during render.